File: CourseScraperPython/Course.py
class Course:
    name = None
    description = None
    credit_hours = None
    gen_ed_categories = list()
    section_list = list()
    section_types = list()

    def __init__(self, json_dict=None):
        if json_dict is None:
            self.name = None
            self.description = None
            self.credit_hours = None
            self.gen_ed_categories = list()
            self.section_list = list()
            self.section_types = list()
        else:
            self.name = json_dict['name']
            self.description = json_dict['description']
            self.credit_hours = json_dict['credit_hours']
            self.gen_ed_categories = json_dict['gen_ed_categories']
            self.section_types = json_dict['section_types']

    def course_to_string(self):
        output = self.name + "\n"
        output += self.description + "\n"
        output += "credit hours: " + str(self.credit_hours) + "\n"
        output += "Gen ed categories: "
        if len(self.gen_ed_categories) <= 0:
            output += "None\n"
        for category in self.gen_ed_categories:
            output += category + " "
        output += "\n"

        for section_type in self.section_types:
            output += section_type + " "
        output += "\n"
        for section in self.section_list:
            output += section.section_to_string() + "\n"
        return output

    def reprJSON(self):
        return dict(name = self.name, description=self.description, credit_hours=self.credit_hours,
                    gen_ed_categories=self.gen_ed_categories, section_list=self.section_list, section_types=self.section_types)

File: CalendarFrontend/src/sample/schedule/Course.java
package sample.schedule;

import com.google.gson.annotations.SerializedName;

import java.util.ArrayList;
import java.util.Arrays;

public class Course {
    @SerializedName("name")
    private String name;

    @SerializedName("description")
    private String description;

    @SerializedName("credit_hours")
    private int creditHours;

    @SerializedName("gen_ed_categories")
    private String[] genEdReqs;

    @SerializedName("section_types")
    private String[] sectionTypes;

    @SerializedName("section_list")
    private Section[] sections;

    private transient ArrayList<String> requiredSections;

    public Course(Course other) {
        this.name = other.name;
        this.description = other.description;
        this.creditHours = other.creditHours;
        this.genEdReqs = other.genEdReqs;
        this.sections = new Section[0];
        this.requiredSections = new ArrayList<>();
    }

    public void addRequiredSection(String newSection) {
        if (requiredSections != null) {
            requiredSections.add(newSection);
        }else{
            requiredSections = new ArrayList<>();
            requiredSections.add(newSection);
        }
    }

    @Override
    public String toString() {
        return "Course{" +
                "name='" + name + '\'' +
                ", description='" + description + '\'' +
                ", creditHours=" + creditHours +
                ", genEdReqs=" + Arrays.toString(genEdReqs) +
                ", sections=" + sections +
                '}';
    }

    public String sectionTypesToString() {
        if (sectionTypes.length > 0) {
            StringBuilder output = new StringBuilder();
            for (int i = 0; i < sectionTypes.length - 1; i++) {
                String type = sectionTypes[i];
                output.append(type);
                output.append(", ");
            }
            output.append(sectionTypes[sectionTypes.length-1]);
            return output.toString();
        }else{
            return "None.";
        }
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public int getCreditHours() {
        return creditHours;
    }

    public void setCreditHours(int creditHours) {
        this.creditHours = creditHours;
    }

    public String[] getGenEdReqs() {
        return genEdReqs;
    }

    public void setGenEdReqs(String[] genEdReqs) {
        this.genEdReqs = genEdReqs;
    }

    public Section[] getSections() {
        return sections;
    }

    public void setSections(Section[] sections) {
        this.sections = sections;
    }

    public String[] getSectionTypes() {
        return sectionTypes;
    }

    public void setSectionTypes(String[] sectionTypes) {
        this.sectionTypes = sectionTypes;
    }

    public ArrayList<String> getRequiredSections() {
        return requiredSections;
    }

    public void setRequiredSections(ArrayList<String> requiredSections) {
        this.requiredSections = requiredSections;
    }
}
